draft-ietf-tls-esni.md

```diff
@@ -603,6 +606,26 @@ length.
 Backend servers (as described in {{server-behavior}}) MUST support the
 "ech_is_inner" extension.
 
+### GREASE PSK {#grease-psk}
+
+When offering ECH, the client is not permitted to advertise PSK identities in
+the ClientHelloOuter. However, it is useful to allow the client to add a fake
+"pre_shared_key" extension to the ClientHelloOuter whenever it advertises PSK
+identities in the ClientHelloInner. This provides cover traffic for session
+resumption with the backend server. When resuming a session with the client, the
+backend server sends a "pre_shared_key" extension in its ServerHello. This would
```

Comment:
```suggestion
the ClientHelloOuter. However, the client can send a "pre_shared_key" extension in the
ClientHelloInner. In this case, when resuming a session with the client, the backend 
server sends a "pre_shared_key" extension in its ServerHello. This would
```

Reply:
(proposed simplification to focus on only extension solicitation violation, rather than on cover traffic, which is the main motivation for this change.)

Reply:
+1. #408 would be another example of a non-cover-traffic motivation which is actually about the real server and not network observers. Worrying about cover traffic suggests needing a threat model and such. The threat model for this is especially weird because the cover traffic can be distinguished with an active client probe:

1. Observe a ClientHello with (possibly GREASE) ECH and (possibly GREASE) PSK.
2. Call the client's bluff and construct a ServerHello that attempts to resume the PSK.
3. See if the client rejects the message.

Reply:
Done.